pyproject.toml

```diff
@@ -9,7 +9,7 @@ name = "cleanlab"
 # requirements files see:
 # https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/
 dependencies = [
-  "numpy>=1.22.0",
+  "numpy~=1.22",
```

Comment:
Numpy was recently updated to 2.0.0, which may fail some unrelated tests.

Reply:
We can revert this if we want before merging.

Reply:
Isn't the near-term plan going forward to support 

numpy >= 1.22.0 < 2.0.0

?

Not sure I understand this syntax you're using here, doesn't seem equivalent to me.

Reply:
I'm also thinking the near-term plan should include supporting version 1.X, specifically 1.22 and above.

---

`~1.X` is equivalent to `>= 1.X, == 1.*`.
https://peps.python.org/pep-0440/#compatible-release

I believe this is a reasonable version specifier as it captures the desired compatibility range.

Reply:
how is `>= 1.X, == 1.*.` equivalent to `>= 1.22.0 < 2.0.0` ?

Doesn't your version include 1.18, 1.19, etc? How do we know those work with cleanlab?

Reply:
let's just make this `>= 1.22.0 < 2.0.0` unless you have a specific reason not to

Reply:
Sorry for any confusion. I thought explaining the pattern with X would make it clearer.

The current change in the PR specifies ~=1.22, which is equivalent to >= 1.22.0 < 2.0.0. This ensures we don't accidentally make a change like >= 1.22.0 < 2.0.1, which would include 2.0.0 in the range. Semantically, this is safer than setting both ranges when we intend to only support versions compatible with 1.22.

---

To clarify, my version does **not** include 1.18, 1.19, etc. It includes 1.22, 1.23, 1.24, 1.25, and 1.26 as of this writing.